app/services/idv/actions/verify_document_status_action.rb

```diff
@@ -0,0 +1,58 @@
+module Idv
+  module Actions
+    class VerifyDocumentStatusAction < Idv::Steps::VerifyBaseStep
+      def call
+        process_async_state(async_state)
+      end
+
+      private
+
+      def process_async_state(current_async_state)
+        render_json case current_async_state.status
+                    when :none
+                      { success: true, status: nil }
+                    when :in_progress
+                      { success: true, status: 'in_progress' }
+                    when :timed_out
+                      { success: false, status: nil, errors: ['timeout'] }
+                    when :done
+                      success = process_result(current_async_state.result)
+                      async_state_done if success
+                      { success: true, status: success ? 'success' : 'fail' }
+                    end
+      end
+
+      def async_state_done
+        delete_async
+        mark_step_complete(:document_capture)
+        save_proofing_components
+      end
+
+      def process_result(result)
+        add_cost(:acuant_result) if result.to_h[:billed]
+        response = idv_result_to_form_response(result)
+        response.success?
+      end
+
+      def async_state
+        dcs_uuid = flow_session[verify_document_capture_session_uuid_key]
+        dcs = DocumentCaptureSession.find_by(uuid: dcs_uuid)
+        return ProofingDocumentCaptureSessionResult.none if dcs_uuid.nil?
+        return ProofingDocumentCaptureSessionResult.timed_out if dcs.nil?
+
+        proofing_job_result = dcs.load_proofing_result
+        return ProofingDocumentCaptureSessionResult.timed_out if proofing_job_result.nil?
+
+        if proofing_job_result.result
+          proofing_job_result.done
```

Comment:
Is this supposed to be...

```suggestion
          ProofingDocumentCaptureSessionResult.done
```

?

I see `done` in the definition of `ProofingDocumentCaptureSessionResult`:

https://github.com/18F/identity-idp/blob/fdb8b416501aba2015ff6d4838e5f86ff17e85e2/app/services/proofing_document_capture_session_result.rb#L21

But not sure where `proofing_job_result.done` would be coming from.

Reply:
> Is this supposed to be...

Disregard this. I was eventually able to follow the logic. I think I hadn't yet come up-to-date with some of the async flow refactorings to `DocumentCaptureSession`.